### PR TITLE
Added Iris Rev6a layout and Milk 2% layout

### PR DESCRIPTION
**Pull request by JellyTitan**
## Description
Added an Iris6a layout with advanced programming examples. Specifically, examples for rotary encoders and per-key RGB.

Added fancy copy/paste layout for milk 2%

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/iris/keymaps/two_knob/keymap.c [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+/**
+ * Rotary Encoder.
+ *
+ * This can't be programmed through configurator. You must do it here.
+ *
+ * Advanced guide for rotary coder programming found here:
+ * https://docs.splitkb.com/hc/en-us/articles/360010513760-How-can-I-use-a-rotary-encoder-
+ */
+bool encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/two_knob/keymap.c:206]: Just a heads up, you may want to check out the Encoder Map feature.  It works much like the keymap, including support for layers, etc.  Given some of your code... it looks like it's exactly what you want to do. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

It also supports all internal keycodes, as well as custom macros!

Comment (JellyTitan) [reply keyboards/keebio/iris/keymaps/two_knob/keymap.c]: Good advice! I refactored accordingly.

File: keyboards/keebio/iris/keymaps/two_knob/config.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+// MOUSEKEY_WHEEL_ACCELERATED_MOVEMENTS  48  Accelerated wheel movements
+// MOUSEKEY_WHEEL_DECELERATED_MOVEMENTS  8 Decelerated wheel movements
+
+/* Used for debugging. Switching these off saves space. */
+#ifndef NO_DEBUG
+#define NO_DEBUG
+#endif /* !NO_DEBUG */
+#if !defined(NO_PRINT) && !defined(CONSOLE_ENABLE)
+#define NO_PRINT
+#endif /* !NO_PRINT */
```
[the remark this answers is not in the document]
Comment (JellyTitan) [reply keyboards/keebio/iris/keymaps/two_knob/config.h]: Good catch - i switched off the console and it saved me almost 5%.

File: keyboards/keebio/iris/keymaps/two_knob/keymap.c [hunk 3]
```diff
@@ -0,0 +1,307 @@
+#include QMK_KEYBOARD_H
```
[the remark this answers is not in the document]
Comment (JellyTitan) [reply keyboards/keebio/iris/keymaps/two_knob/keymap.c:4]: Done!

File: keyboards/keebio/iris/keymaps/two_knob/keymap.c [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 3
```
[the remark this answers is not in the document]
Comment (JellyTitan) [reply keyboards/keebio/iris/keymaps/two_knob/keymap.c]: Done!

File: keyboards/keebio/iris/keymaps/two_knob/keymap.c [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+                    tap_code16(C(KC_TAB));
+                } else {
+                    /* Switch between tabs anticlockwise. (Control + Shift + Tab). */
+                    tap_code16(S(C(KC_TAB)));
+                }
+                break;
+        }
+    } else if (index == 1) {
+        /* Check which layer is active. */
+        switch (biton32(layer_state)) {
```
[the remark this answers is not in the document]
Comment (JellyTitan) [reply keyboards/keebio/iris/keymaps/two_knob/keymap.c]: Done!

File: keyboards/keebio/iris/keymaps/two_knob/keymap.c [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+ * This can't be programmed through configurator. You must do it here.
+ *
+ * Advanced guide for rotary coder programming found here:
+ * https://docs.splitkb.com/hc/en-us/articles/360010513760-How-can-I-use-a-rotary-encoder-
+ */
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    /* index 0 is left encoder. (If you only have one knob, this will be it). */
+    if (index == 0) {
+        /*  Check which layer is active. */
+        switch (biton32(layer_state)) {
```
[the remark this answers is not in the document]
Comment (JellyTitan) [reply keyboards/keebio/iris/keymaps/two_knob/keymap.c]: Done!

File: keyboards/spaceman/2_milk/keymaps/copypasta_macfancy/config.h [hunk 7]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 Ryan Neff (@JellyTitan)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#define TAPPING_TERM 200
```
Comment (drashna) [inline keyboards/spaceman/2_milk/keymaps/copypasta_macfancy/config.h:4]:
```suggestion
#pragma once

#define TAPPING_TERM 200
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keebio/iris/keymaps/two_knob/keymap.c [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QWERTY:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY);
+            }
+            return false;
+            break;
```
Comment (drashna) [inline keyboards/keebio/iris/keymaps/two_knob/keymap.c:99]: This can be removed, since there is no other default base layer

Comment (JellyTitan) [reply keyboards/keebio/iris/keymaps/two_knob/keymap.c]: Redundant case has been removed.

File: keyboards/spaceman/2_milk/keymaps/copypasta_macfancy/config.h (hunk 7 above)
Comment (JellyTitan) [reply keyboards/spaceman/2_milk/keymaps/copypasta_macfancy/config.h]: Added pragma and moved tapping term below

**Review by drashna — APPROVED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Comment by keyboard-magpie** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Pull request merged**